Concurrency 3.1 - Add virtual and qualifier elements to resource definitions

KyleAure opened the pull request:
Fixes #43 

NOTE: testing locally and via GitHub actions resulted in the following error: 
```
      [java] testing application-concurrency.xml
     [java] sax error: 46:cvc-complex-type.2.4.a: Invalid content was found starting with element 'virtual'. One of '{"https://jakarta.ee/xml/ns/jakartaee":qualifier, "https://jakarta.ee/xml/ns/jakartaee":property}' is expected.
     [java] sax error: 53:cvc-complex-type.2.4.a: Invalid content was found starting with element 'virtual'. One of '{"https://jakarta.ee/xml/ns/jakartaee":qualifier, "https://jakarta.ee/xml/ns/jakartaee":property}' is expected.
     [java] sax error: 65:cvc-complex-type.2.4.a: Invalid content was found starting with element 'virtual'. One of '{"https://jakarta.ee/xml/ns/jakartaee":qualifier, "https://jakarta.ee/xml/ns/jakartaee":property}' is expected.
     [java] found 3 error(s)
```

Not sure what this failure means. If someone more familiar with this project could point me in the right direction it would be appreciated. 

(Also weird that errors don't cause build breaks 🤷🏼 )

starksm64 commented on xml/test/application-concurrency.xml:
```diff
@@ -34,17 +34,23 @@
     <context-service-ref>java:app/concurrent/SecContextOnly</context-service-ref>
     <max-async>20</max-async>
+    <qualifier>ee.jakarta.MyQualifier</qualifier>
+    <qualifier>ee.jakarta.MyQualifier2</qualifier>
```
The ordering of the elements in the application_11.xsd for the managed-executorType has the virtual element before the qualifier element, which is why you are seeing validation errors.

KyleAure commented:
@starksm64 thanks for catching that!  I updated the test to have elements in the correct order.

I also updated the web-app-concurrency test which required me to copy/rename the web-app related schemas.

I also want to update the ejb-jar-concurrency test, which would mean we'd have to update the ejb-jar schema to change the following to EE 11: 
https://github.com/jakartaee/jakartaee-schemas/blob/efde9133c2f84ba0ce31fdf775278657a1da4742/xml/src/ejb-jar_4_0.xsds#L74

However there are no new versions of EJB for EE 10/11, so how can embedded resources within this schema be included?
As it stands right now the ejb-jar-concurrency test is failing with 125 errors because it includes elements from EE 10 that the schema doesn't include.

starksm64 commented:
> I also want to update the ejb-jar-concurrency test, which would mean we'd have to update the ejb-jar schema to change the following to EE 11:
> 
> https://github.com/jakartaee/jakartaee-schemas/blob/efde9133c2f84ba0ce31fdf775278657a1da4742/xml/src/ejb-jar_4_0.xsds#L74
> 
> However there are no new versions of EJB for EE 10/11, so how can embedded resources within this schema be included? As it stands right now the ejb-jar-concurrency test is failing with 125 errors because it includes elements from EE 10 that the schema doesn't include.
> 
> Reading some more issues it seems the determination is that EJB schema is frozen at EE 9. Therefore I removed the ejb-jar-concurrency test because it is invalid.
> 
> FYI @tbitonti - This may be of interest to you based on your issue #32

So we won't be able to update the ejb-jar-concurrency.xml test since we don't have any updated schema and there will be no requirements for EJB containers to support the new environments.

njr-11 commented:
We should consider positioning the new `qualifier` element right after `name`.  It might make sense to have this be more prominent in the configuration because it covers how applications reference/access these resources.

edbratt commented:
[How to Publish XML Schemas](https://jakartaee.github.io/platform/publish-xml-schemas) -- Sorry for the wrong link. It seems to incorrect in several places. I'll update the templates  (and a few other places where I noticed this).
